spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/AbstractCassandraJoin.scala

```diff
@@ -0,0 +1,186 @@
+    require(
+      missingPartitionKeys.isEmpty,
+      s"Can't join without the full partition key. Missing: [ $missingPartitionKeys ]"
+    )
+
+    joinColumnNames.foreach(checkSingleColumn)
+    joinColumnNames
+  }
+
+  //We need to make sure we get selectedColumnRefs before serialization so that our RowReader is
+  //built
+  lazy val singleKeyCqlQuery: (String) = {
+    val whereClauses = where.predicates.flatMap(CqlWhereParser.parse)
+    val joinColumns = joinColumnNames.map(_.columnName)
+    val joinColumnPredicates = whereClauses.collect {
+      case EqPredicate(c, _) if joinColumns.contains(c) => c
+      case InPredicate(c) if joinColumns.contains(c) => c
+      case InListPredicate(c, _) if joinColumns.contains(c) => c
+      case RangePredicate(c, _, _) if joinColumns.contains(c) => c
+    }.toSet
+
+    require(
+      joinColumnPredicates.isEmpty,
+      s"""Columns specified in both the join on clause and the where clause.
+          |Partition key columns are always part of the join clause.
+          |Columns in both: ${joinColumnPredicates.mkString(", ")}""".stripMargin
+    )
+
+    logDebug("Generating Single Key Query Prepared Statement String")
+    logDebug(s"SelectedColumns : $selectedColumnRefs -- JoinColumnNames : $joinColumnNames")
+    val columns = selectedColumnRefs.map(_.cql).mkString(", ")
+    val joinWhere = joinColumnNames.map(_.columnName).map(name => s"${quote(name)} = :$name")
+    val limitClause = limit.map(limit => s"LIMIT $limit").getOrElse("")
+    val orderBy = clusteringOrder.map(_.toCql(tableDef)).getOrElse("")
+    val filter = (where.predicates ++ joinWhere).mkString(" AND ")
+    val quotedKeyspaceName = quote(keyspaceName)
+    val quotedTableName = quote(tableName)
+    val query =
+      s"SELECT $columns " +
+        s"FROM $quotedKeyspaceName.$quotedTableName " +
+        s"WHERE $filter $limitClause $orderBy"
+    logDebug(s"Query : $query")
+    query
+  }
+
+  private def boundStatementBuilder(session: Session): BoundStatementBuilder[L] = {
+    val protocolVersion = session.getCluster.getConfiguration.getProtocolOptions.getProtocolVersion
+    val stmt = session.prepare(singleKeyCqlQuery).setConsistencyLevel(consistencyLevel)
+    new BoundStatementBuilder[L](rowWriter, stmt, where.values, protocolVersion = protocolVersion)
+  }
+
+  /**
+   * When computing a CassandraPartitionKeyRDD the data is selected via single CQL statements
+   * from the specified C* Keyspace and Table. This will be preformed on whatever data is
+   * available in the previous RDD in the chain.
+   */
+  override def compute(split: Partition, context: TaskContext): Iterator[(L, R)] = {
+    val session = connector.openSession()
+    val bsb = boundStatementBuilder(session)
+    val metricsUpdater = InputMetricsUpdater(context, readConf)
+    val rowIterator = fetchIterator(session, bsb, left.iterator(split, context))
+    val countingIterator = new CountingIterator(rowIterator, limit)
```

Comment:
change limit to None here, this makes the limit per C* partition instead of per Spark Partition, SPARKC-433